pkg/pillar/zboot/zboot.go

```diff
@@ -360,10 +360,15 @@ func WriteToPartition(log *base.LogObject, image string, partName string) error
 		return errors.New(errStr)
 	}
 
+	// Make sure we have nothing mounted on the target
+	for {
+		if err := syscall.Unmount(devName, 0); err != nil {
+			break
+		}
+		log.Warnf("Successfully umounted %s", devName)
+	}
```

Comment:
Hmm.. You are just trying to unmount multiple times till you receive an error? Can there be other errors for unmount? Like the device being busy? This has the potential to go into infinite loop - isn't it?

Reply:
How can it go in an infinite loop if we get any error like resource busy etc
(We are breaking loop on the first occurrence of the error)

Reply:
I didn't get why you need a loop here? Even if it succeeds - just try once more till u get an error ?
But then, you are not logging the error or not checking if this is the kind of error you want - right?